src/components/afm/tests/utils/DummyHeadlineExecuteAfmAdapter.tsx

```diff
@@ -0,0 +1,24 @@
+// (C) 2007-2018 GoodData Corporation
+import { ISdk } from 'gooddata';
+import { DataSource, ExecuteAfmAdapter } from '@gooddata/data-layer';
+import { AFM, Execution } from '@gooddata/typings';
+import { executionResponses } from './dummyHeadlineFixture';
+
+export class DummyHeadlineExecuteAfmAdapter extends ExecuteAfmAdapter {
+    public createDataSource(afm: AFM.IAfm,
+                            fingerprint?: string): Promise<DataSource.IDataSource<Execution.IExecutionResponses>> {
+        const execFactory = () => {
+            return Promise.resolve(executionResponses);
+        };
+        const dataSource = new DataSource.DataSource<Execution.IExecutionResponses>(
+            execFactory,
+            afm,
+            fingerprint
+        );
+        return Promise.resolve(dataSource);
+    }
+}
+
+export function dummyHeadlineExecuteAfmAdapterFactory(sdk: ISdk, projectId: string) {
```

Comment:
https://en.wikipedia.org/wiki/Factory_method_pattern please change the name its not factory by definition.